src/fhirtypes/ValueSet.ts

```diff
@@ -0,0 +1,72 @@
+import { Meta } from './specialTypes';
+
+import { Extension } from '../fshtypes';
+import { Narrative, Resource, Identifier, CodeableConcept, Coding } from './dataTypes';
+import { ContactDetail, UsageContext } from './metaDataTypes';
+
+/**
+ * Class representing a FHIR R4 ValueSet.
+ *
+ * @see {@link http://hl7.org/fhir/R4/valueset.html | FHIR ValueSet}
+ */
+
+export class ValueSet {
+  id: string;
+  meta: Meta;
+  implicitRules: string;
+  language: string;
+  text: Narrative;
+  contained: Resource[];
+  extension: Extension[];
+  modifierExtension: Extension[];
+  url: string;
+  identifier: Identifier[];
+  version: string;
+  name: string;
+  title: string;
+  status: string;
+  experimental: boolean;
+  date: string;
+  publisher: string;
+  contact: ContactDetail[];
+  description: string;
+  useContext: UsageContext[];
+  jurisdiction: CodeableConcept[];
+  immutable: boolean;
+  purpose: string;
+  copyright: string;
+  compose: ValueSetLogicalDefinition;
+}
+
+export type ValueSetLogicalDefinition = {
+  lockedDate?: string;
+  inactive?: boolean;
+  include: ValueSetCompose[];
+  exclude?: ValueSetCompose[];
```

Comment:
Ah...  I now see the method to your madness...  Still, I like the idea of sticking to existing conventions as much as possible, so even though it may seem strange, I'd suggest this should also use the `ValueSetComposeInclude` type.  Or if that really doesn't sit well with you, we could use the convention as inspiration and call that type `ValueSetComposeIncludeOrExclude`.